### PR TITLE
[#4] Set up CD for the Android app

### LINKED ISSUE #4: Set up CD for the Android app
# Why

We need to deploy the staging and production Android apps automatically to Firebase whenever a PR is merged to `develop` and `main` respectively.

# Acceptance Criteria

Whenever a pull request is merged to `develop` and `main`, the `staging` and `production` apps, respectively, must be deployed to Firebase App Distribution.

# Design
N/A

### PR DESCRIPTION
**Pull request by AVI5HEK**
- Close #4 

## What happened 👀

Whenever a pull request is merged to `develop` and `main`, the staging and production Android apps, respectively, must be deployed to Firebase App Distribution.

## Insight 📝

* Set up the project and apps on Firebase.
* Added the secrets to Github and used them in the existing workflow files created from `flutter-templates`.
* Removed the workflow file to deploy the Android app to the app store since it's out of scope.
* The workflow files for the iOS app are left as-is.

### Successful workflow runs:
* [Staging](https://github.com/nimblehq/ic-flutter-avishek/actions/runs/5420752392/jobs/9855299884).
* [Production](https://github.com/nimblehq/ic-flutter-avishek/actions/runs/5420752393/jobs/9855299883).

#### Staging
image
#### Production
image

### CODE REVIEW

**Comment by markgravity** [comment]
@AVI5HEK can you can the POW please? 🙏 
A screenshot to provide the app is deployed to Firebase and the link to the Github Action also good as well

**Comment by AVI5HEK** [comment]
> @AVI5HEK can you can the POW please? 🙏 A screenshot to provide the app is deployed to Firebase and the link to the Github Action also good as well

Updated. Please check now.

File: .github/workflows/android_deploy_production.yml [hunk 1]
```diff
@@ -34,9 +34,9 @@ jobs:
 
       - name: Set up .env
         env:
-          ENV_PRODUCTION: $
+          ENV_PRODUCTION: ${{ secrets.ENV_PRODUCTION }}
         run: |
-          echo $ENV_PRODUCTION > .env
+          echo $ENV_PRODUCTION | base64 --decode > .env
```
Comment (Wadeewee) [inline .github/workflows/android_deploy_production.yml:39]: @AVI5HEK, You keep data with encoding `base64`, right? Why do you need to `decode` here 🤔

**Review by Wadeewee — COMMENTED** [review]
For the rest LGTM 🚀

File: .github/workflows/android_deploy_production.yml (hunk 1 above)
Comment (AVI5HEK) [reply .github/workflows/android_deploy_production.yml:39]: @Wadeewee The file is saved in the secrets as a base64 string. So we need to decode it back to the `.env` file before building the app.

Comment (Wadeewee) [reply .github/workflows/android_deploy_production.yml:39]: Thanks for the confirmation @AVI5HEK 🙏

**Review by Wadeewee — APPROVED** [review]

**Review by markgravity — APPROVED** [review]

**Pull request merged**